In [11]:
import pandas as pd
import statsmodels.api as sm

In [9]:
data = {
    "y": [180, 98.1, 173.1, 136.5, 141, 165.9, 193.5, 127.8, 163.5, 172.5],
    "x": [23, 11, 20, 17, 15, 21, 24, 13, 19, 25]
}
df = pd.DataFrame(data)

In [12]:
X = sm.add_constant(df["x"])
model = sm.OLS(df["y"], X).fit()

x0 = 20
point_estimate = model.predict([1, x0])[0]
point_estimate

np.float64(162.03035714285707)

In [13]:
pred = model.get_prediction([1, x0])
ci_mean = pred.conf_int(alpha=0.05)

ci_mean

array([[154.03837651, 170.02233778]])

95% conf interval [154.03837651, 170.02233778] for mean price of all houses with size 20